src/Driver/Driver.php

```
@@ -689,19 +689,51 @@ protected function rollbackSavepoint(int $level): void
         $this->execute('ROLLBACK TO SAVEPOINT ' . $this->identifier("SVP{$level}"));
     }
 
+    /**
+     * @return array{string, string, string}
+     */
+    private function parseDSN(): array
+    {
+        $dsn = $this->getDSN();
+
+        $user = (string)($this->options['username'] ?? '');
+        $pass = (string)($this->options['password'] ?? '');
+
+        if (\strpos($dsn, '://') > 0) {
+            $parts = \parse_url($dsn);
+
+            // Update username and password from DSN if not defined.
+            $user = $user ?: $parts['user'] ?? '';
+            $pass = $pass ?: $parts['pass'] ?? '';
+
+            // Build new DSN
+            $dsn = \vsprintf('%s:host=%s', [
+                $parts['scheme'] ?? 'mysql',
```

I think, if the scheme value not exists then we should throw an exception here